src/Ast/PhpDoc/AssertTagMethodValueNode.php

```diff
@@ -23,23 +23,28 @@ class AssertTagMethodValueNode implements PhpDocTagValueNode
 	/** @var bool */
 	public $isNegated;
 
+	/** @var bool */
+	public $isEquality;
+
 	/** @var string (may be empty) */
 	public $description;
 
-	public function __construct(TypeNode $type, string $parameter, string $method, bool $isNegated, string $description)
+	public function __construct(TypeNode $type, string $parameter, string $method, bool $isNegated, string $description, bool $isEquality = false)
 	{
 		$this->type = $type;
 		$this->parameter = $parameter;
 		$this->method = $method;
 		$this->isNegated = $isNegated;
+		$this->isEquality = $isEquality;
 		$this->description = $description;
 	}
 
 
 	public function __toString(): string
 	{
 		$isNegated = $this->isNegated ? '!' : '';
-		return trim("{$isNegated}{$this->type} {$this->parameter}->{$this->method}() {$this->description}");
+		$isEquality = $this->isEquality ? '=' : '';
+		return trim("{$isNegated}{$isEquality}{$this->type} {$this->parameter}->{$this->method}() {$this->description}");
```

Comment:
seeing this, I am wondering whether paranthesis as I used them in

https://github.com/phpstan/phpstan-src/pull/1880/files#diff-8f0c25b196228aa9450cbd3c1d8c83d6d5f33c9fe0aa1b07effe32d41275dd3eR41

are a valid thing?

Reply:
They're valid since you can just parenthesize any type, but it's not necessary.

Reply:
additionally, maybe its worth having a test-case for e.g. `@phpstan-assert-if-false !=(float|int|numeric-string) $value`